src/main/kotlin/no/nav/sykdig/db/OppgaveRepository.kt

```diff
@@ -67,19 +72,95 @@ class OppgaveRepository(private val namedParameterJdbcTemplate: NamedParameterJd
                                                                                              WHERE s.oppgave_id = o.oppgave_id
                                                                                              ORDER BY timestamp DESC
                                                                                              LIMIT 1)
-                    where o.oppgave_id = :oppgave_id;
+                    WHERE o.oppgave_id = :oppgave_id;
             """,
             mapOf("oppgave_id" to oppgaveId)
         ) { resultSet, _ ->
             resultSet.toDigitaliseringsoppgave()
         }.firstOrNull()
     }
 
+    fun updateOppgave(
+        oppgaveId: String,
+        values: SykmeldingUnderArbeidValues,
+    ) {
+        val sykmeldingUnderArbeid = getOppgave(oppgaveId).let {
+            requireNotNull(it) { "Oppgave can't not exist at this point" }
+            it.sykmelding
+        }.updateOrCreateSykmeldingUnderArbeid(values);
+
+        namedParameterJdbcTemplate.update(
+            """
+                UPDATE oppgave
+                SET fnr = :fnr
+                WHERE oppgave_id = :oppgave_id
+            """.trimIndent(),
+            mapOf(
+                "oppgave_id" to oppgaveId,
+                "fnr" to values.fnrPasient,
+            )
+        )
+        namedParameterJdbcTemplate.update(
+            """
+                UPDATE sykmelding
+                SET sykmelding = :sykmelding
+                WHERE sykmelding_id = :sykmelding_id
+            """.trimIndent(),
+            mapOf(
+                "sykmelding_id" to sykmeldingUnderArbeid.sykmelding?.id,
+                "sykmelding" to sykmeldingUnderArbeid.toPGObject(),
+            )
+        )
+    }
+
     fun deleteAll() {
         namedParameterJdbcTemplate.update("DELETE FROM sykmelding", MapSqlParameterSource())
     }
 }
 
+private fun SykmeldingUnderArbeid?.updateOrCreateSykmeldingUnderArbeid(
```

Comment:
Dette er vel egentlig bare en oppretting eller oppdatering av objektet for SykmeldingUnderArbeid, den gjør ikke noe med databasen? Jeg blir litt forvirret når den heter UpdateOrCreate og ligger i repository-klassen uten at den gjør noe mot databasen. Kanskje den burde flyttes til en egen klasse/fil? :)